src/Headers/Hydrodynamics.h

```diff
@@ -84,6 +86,9 @@ class Hydrodynamics
   void CheckBoundaryGhostParticle(const int, const int, const FLOAT, const DomainBox<ndim> &);
 
   void CreateBoundaryGhostParticle(const int, const int, const int, const FLOAT, const FLOAT);
+  Particle<ndim>& CreateNewParticle(const enum ptype, const enum parttype, const int, const int, const int,
+                                    const FLOAT, const FLOAT, const FLOAT, const FLOAT*, const FLOAT*,SimulationBase*);
+
```

Comment:
Is this pointer to SimulationBase really necessary? I know we use it to flag for a tree rebuild, but I'm not super happy with this solution.